src/cursor/cursor.ts

```diff
@@ -845,24 +920,18 @@ export class Cursor<
     return this.isDead();
   }
 
-  destroy(err?: AnyError): void {
-    if (err) this.emit(Cursor.ERROR, err);
-    this.pause();
-    this.close();
-  }
-
   /** Return a modified Readable stream including a possible transform method. */
-  stream(options?: StreamOptions): this {
+  stream(options?: CursorStreamOptions): CursorStream {
     // TODO: replace this method with transformStream in next major release
     this.streamOptions = options || {};
-    return this;
+    return new CursorStream(this);
```

Comment:
`new CursorStream(this, options)` - let's _not_ store `streamOptions` on the cursor itself, that's a side effect

Reply:
On the subject of `.stream()`, what's with the TODO comment on L925?

`// TODO: replace this method with transformStream in next major release`

Maybe this is the right time to do that? Do you have any context on that TODO?

Reply:
At this point I think we really want to just remove `transformStream` and standardize on `stream`. I left another comment about that, let's tackle it in another ticket.